src/server/unit_sao.cpp

```diff
@@ -124,6 +124,16 @@ void UnitSAO::sendOutdatedData()
 void UnitSAO::setAttachment(int parent_id, const std::string &bone, v3f position,
 		v3f rotation, bool force_visible)
 {
+	auto *obj = parent_id ? m_env->getActiveObject(parent_id) : nullptr;
+	if (obj) {
+		// Do checks to avoid circular references
+		// The chain of wanted parent must not refer or contain "this"
+		for (obj = obj->getParent(); obj; obj = obj->getParent()) {
```

Comment:
we should log if someone try to attach a parent to its child, it's clearly a bug or a modding error which should not be silent

Reply:
warningstream added